format/spec.md

```diff
@@ -221,6 +222,8 @@ The `initial-default` is set only when a field is added to an existing schema. T
 
 The `initial-default` and `write-default` produce SQL default value behavior, without rewriting data files. SQL default value behavior when a field is added handles all existing rows as though the rows were written with the new field's default value. Default value changes may only affect future records and all known fields are written into data files. Omitting a known field when writing a data file is never allowed. The write default for a field must be written if a field is not supplied to a write. If the write default for a required field is not set, the writer must fail.
 
+All columns of `unknown` type must default to null. Non-null values for `initial-default` or `write-default` are invalid.
```

Comment:
I'm a little confused about how physical columns in datafiles should be handled if they are present.  If there is a column with a matching field id and values present, should those be materialized as null as well?

Reply:
Further down, it states that these columns should be omitted from ORC and Parquet files. Avro can keep the type in the schema, but the serialized format ends up being the same.

Reply:
It probably makes sense to update the column projection rules to specify this type shouldn't be matched?  At least for old data, parquet supports UNKNOWN logical types as well.

Reply:
What do you mean? How would this type be matched in column projection? If we're reading a Parquet file with a matching column ID, then it is invalid because you can't write these columns to Parquet. I don't think that we would specifically ignore that case for an `unknown` column. Maybe it should be an error?

Reply:
@rdblue I think one could potentially import parquet files that have an existing column annotated as a logical type "Unknown".  I'd expect for these files the inferred schema would be the unknown iceberg type as well.  So they could in theory be read from the files (via column name mapping) but we potentially want to skip them, do avoid any complication.

Reply:
I've added a note in the Parquet Appendix that Parquet columns that correspond to `unknown` columns must be ignored and replaced with `null`. I think that's the cleanest location for this.